### PR TITLE
NOISSUE - Add Invitations Service

### LINKED ISSUE #14: Update JS SDK for invitations

### PR DESCRIPTION
**Pull request by Musilah**
### What does this do?
This adds an Invitations Service to the SDK.
### Which issue(s) does this PR fix/relate to?
Resolves #14 
### List any changes that modify/break current functionality
Add TS invitations svc for sdk
### Have you included tests for your changes?
NO
### Did you document any new/modified functionality?
NO

### CODE REVIEW

File: src/defs.ts [hunk 1]
```diff
@@ -139,3 +139,16 @@ export interface DomainsPage {
 export interface Permissions {
   permissions: string[]
 }
+
+export interface Invitation {
```
Comment (ianmuchyri) [inline src/defs.ts:143]: An invitation struct looks like this:
```
type Invitation struct {
	InvitedBy   string    `json:"invited_by"`
	UserID      string    `json:"user_id"`
	DomainID    string    `json:"domain_id"`
	Token       string    `json:"token,omitempty"`
	Relation    string    `json:"relation,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at,omitempty"`
	ConfirmedAt time.Time `json:"confirmed_at,omitempty"`
	Resend      bool      `json:"resend,omitempty"`
}

```
Add missing fields and remove `metadata` field.

Also update the `User` interface to include `created_at` field

Comment (ianmuchyri) [reply src/defs.ts:143]: also use snake case

File: src/invitations.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+     * relation: '<role>' // available options: 'admin', 'owner', 'viewer', 'editor'
+     * }
+     */
+    const options: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      },
+      body: JSON.stringify({ user_id: invitation.userID, domain_id: invitation.domainID, relation: invitation.relation })
```
Comment (ianmuchyri) [inline src/invitations.ts:49]: Once the issue in defs.ts is solved, you can just use 
```suggestion
      body: JSON.stringify(invitation)
```

File: src/invitations.ts [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    /**
+     * @method SendInvitation - sends an invitation to the email address associated with the given user.
+     * @param {Object} invitation - The invitation object.
+     * @param {string} token - The user's access token.
+     * @returns {Object} - The response object which has a status and a message.
+     * @example
+     * const invitation = {
+     *  userID: '<userID>',
+     *  domainID: '<domainID>',
+     * relation: '<role>' // available options: 'admin', 'owner', 'viewer', 'editor'
```
Comment (ianmuchyri) [inline src/invitations.ts:40]:
```suggestion
     * relation: '<role>' // available options: 'administrator', 'editor','viewer','member'
```

File: src/defs.ts (hunk 1 above)
Comment (ianmuchyri) [reply src/defs.ts:143]: For relation use type `Relation`

File: src/invitations.ts [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        throw this.invitationError.HandleError(errorRes.error, response.status)
+      }
+      const inviteResponse: Response = { status: response.status, message: 'Invitation Sent Successfully' }
+      return inviteResponse
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async Invitation (invitation: Invitation, token: string): Promise<Invitation> {
```
Comment (ianmuchyri) [inline src/invitations.ts:68]:
```suggestion
  public async Invitation (userID: string, domainID: string, token: string): Promise<Invitation> {
```

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

File: src/invitations.ts [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        throw this.invitationError.HandleError(errorRes.error, response.status)
+      }
+      const invitationData: InvitationsPage = await response.json()
+      return invitationData
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async AcceptInvitation (invitation: Invitation, token: string): Promise<Response> {
```
Comment (ianmuchyri) [inline src/invitations.ts:139]:
```suggestion
  public async AcceptInvitation (domainID: string, token: string): Promise<Response> {
```

File: src/invitations.ts [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+     * @param {string} token - The user's access token.
+     * @returns {Object} - The response object which has a status and a message.
+     */
+    const options: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      },
+      body: JSON.stringify({ domain_id: invitation.domain_id })
```
Comment (ianmuchyri) [inline src/invitations.ts:153]:
```suggestion
      body: JSON.stringify({ domain_id: domainID})
```

File: src/invitations.ts [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+        throw this.invitationError.HandleError(errorRes.error, response.status)
+      }
+      const inviteResponse: Response = { status: response.status, message: 'Invitation Accepeted Successfully' }
+      return inviteResponse
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async DeleteInvitation (invitation: Invitation, token: string): Promise<Response> {
```
Comment (ianmuchyri) [inline src/invitations.ts:172]:
```suggestion
  public async DeleteInvitation (userID: string, domainID: string, token: string): Promise<Response> {
```

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

**Review by ianmuchyri — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**